resources/AutomIAlib.py

```diff
@@ -47,3 +46,29 @@ def read_properties_file(file_path:str) -> list[str]:
 
         # Return sorted properties
         return properties
+
+    @staticmethod
+    def update_json_file(file_path: str, key_value: str):
+        """
+        Updates a JSON file by adding or modifying a key-value pair at the root level.
+
+        Arguments:
+        - file_path: Path to the JSON file to be modified.
+        - key_value: String representing the key-value pair in the format "key:value".
+        """
+        # Split the key_value string into key and value
+        key, value = key_value.split(':', 1)
+        # Convert value to int if it's numeric, otherwise keep it as a string
+        value = int(value) if value.isdigit() else value
+
+        # Load the existing JSON data
+        file_path = file_path + ".json"
+        with open(file_path, 'r', encoding="utf8") as file:
+            json_data = json.load(file)
+
+        # Update or add the key-value pair
+        json_data[key] = value
```

Comment:
il faut vérifier que json_data est bien non null sinon on risque le crash

Reply:
Ajout des lignes suivantes :
        # Check if json_data is None
        if json_data is None:
            raise ValueError("The JSON data could not be loaded or is empty.")